### DISCUSSION TITLE
high CPU load while tagging faces

### DISCUSSION THREAD

**Discussion by ippocratis** [photoprism/photoprism#3175] [Q&A]
I'm running photoprism on a ruspberry pi 400 with docker-compose

I've run a
`docker compose photoprism photoprism inex` which indexed my assets fine and after it finished `docker compose exec photoprism photoprism faces index`
which added faces to photos according to logs

Navigating the Webui works fine

But

The if I select the  people section ui freezes

If I monitor the CPU load with `top` I see photoprism and mysql occupying 300-400% (4 core SBC) which leads to an auto reboot of the raspberry (have the hardware watchdog enabled)

**Comment by ippocratis** [comment]
Update on this:

Renamed the issue as it looks like it was a max-load-1 missconfiguration in the watchdog config

Disabling the watchdog don't crash my rpi
Face tagging is slow but doable


Not sure if I have to close the issue as its not a photoprism bug but has to do with finding a more precise Max load value for the watchdog

On the other hand some other users may have a better idea about this

**Reply by lastzero** [reply]
I've moved this to discussions so others can comment on it!

**Reply by lastzero** [reply]
Would you be able to share more specifics on the watchdog you have mentioned (I've never heard about it)?

That sounds like something we should add to the troubleshooting checklists we provide:
- https://docs.photoprism.app/getting-started/troubleshooting/

Of course, you are also welcome to add it on your own and send a pull request:
- https://github.com/photoprism/photoprism-docs/blob/master/docs/getting-started/troubleshooting/index.md

**Reply by ippocratis** [reply]
Watch Dog Timer is an electronic timer used to detect and recover from computer malfunctions.

If Rpi fails to reset the timer before it expires, the WDT signal reboots the computer.

The idea is , for rpis that run 24/7 if the system gets unresponsive,  reboot it 

It is disabled by default in the firmware

https://github.com/raspberrypi/firmware/blob/f694bbe7c6f142e0c1a5033f0f6c15528fd6c98c/boot/overlays/README#L277

Users can set up "dtparam" aka Device Tree configuration files for Raspberry Pi's in 
/boot/config.txt which will enable the kernel module

It is the user's responsibility to set up the watchdog daemon parameters correctly

According to the watchdog.conf man page
https://linux.die.net/man/5/watchdog.conf

One common parameter users setup is the CPU load average for 1,5 or 15 min and the default value is 24

max-load-1 = ?
Set the maximal allowed load average for a 1 minute span
And according to the watchdog man page load average is read from /proc/loadavg

As far as i can tell

The load average is the sum of the run queue length and the number of jobs currently running on the CPUs.

Commands that can provide load average stats are

uptime
procinfo
w
top

I think a proper workaround for raspberry users that have it enabled  (but also for computers in general that use  hardware watchdogs) is a script(?) to export the above commands  timestambed load average  , or maybe directly read from /proc/loadavg and log it to file. Monitoring the highest values while the new faces tagging is taking place and select a higher value so that the WD don't trip while working with faces.

I saw values high as 140 for the 1 minute load 

Also maybe the 5 minute (max-load-5) or the 15 minutes (max-load-15) config values is more appropriate

**Reply by ippocratis** [reply]
I ended up with the below script to log load average into a file.

```
#!/bin/bash

while true; do
echo $(cat /proc/loadavg) >> test_file.log
sleep 10
done
```

I could `echo` any of the above commands like

`echo $(uptime) >> test_file.log`

Which will add a timestamp to the log entries

and select a diferent log interval in seconds  `sleep 10`

**Reply by ippocratis** [reply]
@lastzero please have a look to https://github.com/photoprism/photoprism-docs/pull/153
I've gathered my comments from this discussion  to a pull request as you suggested

**Reply by lastzero** [reply]
Thank you! I will take a look as soon as possible. Need a short break after the release :)
